### ISSUE TITLE
[Regression] input_device_display_name in autoconfig file not being displayed anymore

### ISSUE THREAD

**Issue by Ryunam** [libretro/RetroArch#14211]
## Description

Current RA master is not displaying the `input_device_display_name` value of an autoconfig profile anymore. Instead, the controller notification only shows what is reported as the `input_device`.

I have bisected this and the issue started from fa9e56d824af81196fd0227821121b65f2d77ef7.

### Expected behavior

`input_device_display_name` should be shown in the controller notification, if it is present in the corresponding .cfg autoconfig profile. Otherwise, if that value is entirely absent from the autoconfig file, `input_device` should be used as a fallback label.

### Actual behavior

Currently, `input_device` is always being used as the identifier in the controller notification and `input_device_display_name` is ignored.

### Bisect Results

First bad commit: fa9e56d824af81196fd0227821121b65f2d77ef7 by @LibretroAdmin 

### Version/Commit

- RetroArch: 1.10.3/35d6c31b5b11d30ac0422e1b699a14952a4fa87d

### Environment information

- OS: Windows 11 x64
- Compiler: MSYS2/Mingw64

**Comment by bslenul** [comment]
Weird, I use a DS4 + DS4Windows and I have a custom autoconfig as well with PS labels, proper display name, etc. but for me it still loads the proper autoconfig although the initial popup now says "Microsoft Xbox 360 Controller" instead of "DualShock 4":

![image](https://user-images.githubusercontent.com/33353403/180662170-4e5ce2ba-ce38-4f50-851b-a690158441d8.png)
![image](https://user-images.githubusercontent.com/33353403/180662211-53cde255-e705-4ec2-b747-ff7770508def.png)
![image](https://user-images.githubusercontent.com/33353403/180662195-32f1c83e-ec53-4e27-8a2d-92008b047bf8.png)

**Comment by Ryunam** [comment]
Hmm, initially I thought it might be simply a case of the `input_device_display_name` parameter not being read properly. However, I tried manually editing my custom autoconfig profile and the changes I have made to the file do not reflect in the way RA presents those inputs.

Example: In my autoconfig profile I changed the name of the button assigned to the Menu hotkey, from "Playstation Button" to "Claystation Button". When I run RA, it still detects it as "Playstation Button".

**Comment by bslenul** [comment]
Where does it get that "Playstation Button" label? You checked in "System Information" to see which .cfg file was being used (just like my 3rd screenshot above)?

Edited labels show properly for me:

![image](https://user-images.githubusercontent.com/33353403/180750658-8c7a45c7-1064-4ae7-a928-11b0cd6ee0a9.png)

The only issue I have is the popup not showing the `input_device_display_name` anymore.

**Comment by Ryunam** [comment]
@bslenul You're absolutely right! I eventually figured out why I had thought that the autoconfig detection mechanism was impacted. 

Turns out I had two customized autoconfig profiles in the same folder, mapped to the same VID/PID of the Dualsense+DS4Windows setup and with the same controls/labels, but with two different input_device values. 

Before fa9e56d824af81196fd0227821121b65f2d77ef7 RA used to load one of these two profiles, now it loads the other one.

I can confirm that we can ultimately narrow down this issue to the `input_device_display_name` parameter not being displayed anymore in current RA. I will now edit the name of the issue and its description accordingly.
